Makefile

```
@@ -80,7 +80,7 @@ endif
 
 # (method=thread to support xdist)
 PYTEST_OPTS := -v -p no:logging $(PYTEST_OPTS)
-PYTHONPATH=$(shell pwd)/contrib/pyln-client:$(shell pwd)/contrib/pyln-testing:$(shell pwd)/contrib/pyln-proto/:$(shell pwd)/external/lnprototest:$(shell pwd)/contrib/pyln-spec/bolt1:$(shell pwd)/contrib/pyln-spec/bolt2:$(shell pwd)/contrib/pyln-spec/bolt4:$(shell pwd)/contrib/pyln-spec/bolt7
+MY_CHECK_PYTHONPATH=$(shell pwd)/contrib/pyln-client:$(shell pwd)/contrib/pyln-testing:$(shell pwd)/contrib/pyln-proto/:$(shell pwd)/external/lnprototest:$(shell pwd)/contrib/pyln-spec/bolt1:$(shell pwd)/contrib/pyln-spec/bolt2:$(shell pwd)/contrib/pyln-spec/bolt4:$(shell pwd)/contrib/pyln-spec/bolt7
```

Wouldn't augmenting the `PYTHONPATH` instead not be better?

```suggestion
PYTHONPATH=${PYTHONPATH}:$(shell pwd)/contrib/pyln-client:$(shell pwd)/contrib/pyln-testing:$(shell pwd)/contrib/pyln-proto/:$(shell pwd)/external/lnprototest:$(shell pwd)/contrib/pyln-spec/bolt1:$(shell pwd)/contrib/pyln-spec/bolt2:$(shell pwd)/contrib/pyln-spec/bolt4:$(shell pwd)/contrib/pyln-spec/bolt7
```

This way we could keep OS specific paths in the search, overriding it when we actually need it.

---

A simplified example seems to fail though:

```
$ cat Makefile
PYTHONPATH=${PYTHONPATH}:/example

.PHONY: show-pythonpath

show-pythonpath:
        PYTHONPATH=$(PYTHONPATH) echo $$PYTHONPATH
$ make show-pythonpath
Makefile:1: *** Recursive variable 'PYTHONPATH' references itself (eventually).  Stop.
```

Seems just overall safer to outright use a completely different name.

---

Okay, so `echo $PYTHONPATH` is wrong since the `$PYTHONPATH` gets expanded "early", but still:

```
$ cat Makefile
PYTHONPATH=$$PYTHONPATH:/example

.PHONY: show-pythonpath

show-pythonpath:
        PYTHONPATH=$(PYTHONPATH) env | grep PYTHONPATH
$ make show-pythonpath 
PYTHONPATH=$PYTHONPATH:/example env | grep PYTHONPATH
PYTHONPATH=$PYTHONPATH:/example:/example
```

---

This one seems to work out better, and works compatibly on systems which do *not* set `PYTHONPATH`, either:

```$ cat Makefile
MY_CHECK_PYTHONPATH=$${PYTHONPATH}$${PYTHONPATH:+:}/example

.PHONY: show-pythonpath

show-pythonpath:
        PYTHONPATH=$(MY_CHECK_PYTHONPATH) env | grep PYTHONPATH
$ make show-pythonpath
PYTHONPATH=${PYTHONPATH}${PYTHONPATH:+:}/example env | grep PYTHONPATH
PYTHONPATH=/home/zmnscpxj/.guix-profile/lib/python3.8/site-packages:/example
$ PYTHONPATH= make show-pythonpath
PYTHONPATH=${PYTHONPATH}${PYTHONPATH:+:}/example env | grep PYTHONPATH
PYTHONPATH=/example
```